libs/application/templates/marriage-conditions/src/fields/NationalIdWithName/NationalIdWithName.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { FC, useEffect, useState } from 'react'
+import { Box, GridRow, GridColumn } from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import { getErrorViaPath } from '@island.is/application/core'
+import { FieldBaseProps } from '@island.is/application/types'
+import { gql, useLazyQuery } from '@apollo/client'
+import { IdentityInput, Query } from '@island.is/api/schema'
+import { InputController } from '@island.is/shared/form-fields'
+import { useFormContext } from 'react-hook-form'
+import * as kennitala from 'kennitala'
+import { m } from '../../lib/messages'
+
+const IdentityQuery = gql`
+  query IdentityQuery($input: IdentityInput!) {
```

Comment:
I think you can add this query in a queries section of the application and use the codegen to generate easy to use hooks for you. But this works.